sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -830,6 +830,70 @@ class DataSourceV2SQLSuite
     assert(df.collect().map(_.getAs[String](0)).sorted === expected.sorted)
   }
 
+  test("UseCatalog: use catalog with v2 catalog") {
+    val catalogManager = spark.sessionState.catalogManager
+    assert(catalogManager.currentCatalog.name() == "session")
+
+    sql("USE CATALOG testcat")
+    assert(catalogManager.currentCatalog.name() == "testcat")
+  }
+
+  test("UseCatalog: v2 catalog does not exist") {
+    val exception = intercept[AnalysisException] {
+      sql("USE CATALOG unknown")
+    }
+    assert(exception.getMessage.contains("v2 catalog 'unknown' cannot be loaded"))
+  }
+
+  test("Use: basic tests with USE statements") {
+    val catalogManager = spark.sessionState.catalogManager
+
+    // Validate the initial current catalog and namespace.
+    assert(catalogManager.currentCatalog.name() == "session")
+    assert(catalogManager.currentNamespace === Array("default"))
+
+    // The following implicitly creates namespaces.
```

Comment:
this reminds that we should have a `CREATE NAMESPACE` command. We can do it in followup.

Reply:
Yes, I intend to do it right after this.

Reply:
I created a [JIRA](https://issues.apache.org/jira/browse/SPARK-29278) to track this along with `SHOW CURRENT CATALOG` and `SHOW CURRENT NAMESPACE` as you suggested.